src/main/java/core/basesyntax/service/impl/FruitRawStringParserImpl.java

```diff
@@ -0,0 +1,40 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.dto.FruitTransactionDto;
+import core.basesyntax.exception.DataFileCorrupted;
+import core.basesyntax.service.interfaces.FruitRawStringParser;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FruitRawStringParserImpl implements FruitRawStringParser {
+    private static final int OPERATION_NEXT_INDEX = 0;
+    private static final int FRUIT_NAME_INDEX = 1;
+    private static final int FIRST_CVS_LINE = 1;
+    private static final int FRUIT_QUANTITY_INDEX = 2;
+
+    private FruitStringsCheckImpl checkFruitsInFile = new FruitStringsCheckImpl();

```

Comment:
```suggestion
    private final FruitStringsCheck checkFruitsInFile;

    public FruitRawStringParserImpl(FruitStringsCheckImpl checkFruitsInFile) {
        this.checkFruitsInFile = checkFruitsInFile;
    }

```
Field could be final, it`s better to use abstraction, and i`m almost sure that it is better to create an instance of FruitStringsCheck in main and pass it as the parameter in constuctor.

Reply:
And also, since this check is used only in one class, I think it may be even better to create a private method in the same class rather than a separate interface and class.